File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.js
function solution(numbers) {
    // 1. 숫자들을 문자열로 변환하여 정렬
    const answer = numbers
        .map(n => n.toString())
        .sort((a, b) => (b + a) - (a + b))
        .join('');
    
    // 2. 예외 처리: 모든 숫자가 0인 경우
    return answer[0] === '0' ? '0' : answer;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        final int LEN = numbers.length;
        String[] stringifyNumbers = new String[LEN];
        StringBuilder sb = new StringBuilder();
        
        for(int i = 0; i < LEN; i++){
            stringifyNumbers[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(stringifyNumbers, (a, b) -> (b + a).compareTo(a + b));
        
        for(int i = 0; i < LEN; i++){
            sb.append(stringifyNumbers[i]);
        }
        
        if(stringifyNumbers[0].equals("0")) {
            return "0";
        }
        
        return sb.toString();
    }
}